pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -85,43 +85,38 @@ func (m *wrappedManagerImpl) PluginListAndWatchReceiver(r string, resp *pluginap
 	m.callback(r, devices)
 }
 
-func tmpSocketDir() (socketDir, socketName, pluginSocketName string, err error) {
-	socketDir, err = os.MkdirTemp("", "device_plugin")
-	if err != nil {
-		return
-	}
+func tmpSocketDir(t *testing.T) (socketName, pluginSocketName string) {
+	socketDir, err := os.MkdirTemp("", "device_plugin")
+	require.NoError(t, err)
 	socketName = filepath.Join(socketDir, "server.sock")
 	pluginSocketName = filepath.Join(socketDir, "device-plugin.sock")
 	os.MkdirAll(socketDir, 0755)
+	t.Cleanup(func() {
+		os.RemoveAll(socketDir)
+	})
 	return
 }
 
 func TestNewManagerImpl(t *testing.T) {
-	socketDir, socketName, _, err := tmpSocketDir()
+	socketName, _ := tmpSocketDir(t)
 	topologyStore := topologymanager.NewFakeManager()
+	_, err := newManagerImpl(socketName, nil, topologyStore)
 	require.NoError(t, err)
-	defer os.RemoveAll(socketDir)
-	_, err = newManagerImpl(socketName, nil, topologyStore)
-	require.NoError(t, err)
-	os.RemoveAll(socketDir)
 }
 
 func TestNewManagerImplStart(t *testing.T) {
-	socketDir, socketName, pluginSocketName, err := tmpSocketDir()
-	require.NoError(t, err)
-	defer os.RemoveAll(socketDir)
+	socketName, pluginSocketName := tmpSocketDir(t)
 	m, _, p := setup(t, []*pluginapi.Device{}, func(n string, d []pluginapi.Device) {}, socketName, pluginSocketName)
-	cleanup(t, m, p)
 	// Stop should tolerate being called more than once.
-	cleanup(t, m, p)
+	p.Stop()
+	p.Stop()
+	m.Stop()
+	m.Stop()
```

Comment:
This does add a third call to each of these `Stop()` methods including the one added by `setup()`, though I thought this looked a little less weird in the context of this comment than relying on the implicit `Stop()` calls from `setup()`.